otter/test/test_controller.py

```diff
@@ -1464,3 +1471,96 @@ def test_checks_pass_replace_true_no_purge_failure(self):
         ])
         self.assertRaises(ValueError, self._remove, True, False,
                           seq_dispatcher)
+
+    def test_perform_convergence_remove_from_group(self):
+        """
+        Perform :func:`convergence_remove_server_from_group` with the given
+        dispatcher.
+        """
+        self.state.desired = 2
+        fake_super = FakeSupervisor()
+
+        dispatcher = ComposedDispatcher([
+            test_dispatcher(),
+            SequenceDispatcher([
+                self._tenant_retry(
+                    get_server_details('server_id').intent,
+                    lambda _: (StubResponse(200, {}), self.server_details)),
+                (GetScalingGroupInfo(tenant_id='tenant_id',
+                                     group_id='group_id'),
+                    lambda _: (self.group, self.group_manifest_info)),
+                self._tenant_retry(
+                    mock.ANY,
+                    lambda i: sync_perform(dispatcher, Effect(i)))
+            ]),
+            get_eviction_dispatcher(fake_super)
```

Comment:
Hmm... fair enough.  I intended the original test as more pinning several units together, but now that I've done the refactor this particular is merely testing that the effect is performed with the given dispatcher...